main_test.go

```diff
@@ -394,3 +394,89 @@ func TestProtocol(t *testing.T) {
 	case <-time.After(time.Second):
 	}
 }
+
+func TestTemp(t *testing.T) {
+
+	loggo.ConfigureLoggers("<root>=TRACE")
+
+	priv_key1, _, _ := p2p_crypto.GenerateECDSAKeyPair(rand.Reader)
+	priv_key2, _, _ := p2p_crypto.GenerateECDSAKeyPair(rand.Reader)
+	priv_key3, _, _ := p2p_crypto.GenerateECDSAKeyPair(rand.Reader)
+
+	registerBootstrap(priv_key1, ":41011")
+	registerBootstrap(priv_key2, ":41012")
+	registerBootstrap(priv_key3, ":41013")
+
+	done1 := make(chan bool)
+	done2 := make(chan bool)
+
+	goldenMessage := []byte("dfstrdfgcrjtdg")
+
+	handler1 := func(msg []byte) {
+		log.Infof("received message 1: %s", string(msg))
+		log.Infof("len, %v", len(goldenMessage))
+		log.Infof("len, %v", len(msg))
+		if !bytes.Equal(msg, goldenMessage) {
+			log.Errorf("bad response")
+		}
+		assert.Equal(t, msg, goldenMessage)
+		done1 <- true
+	}
+	handler2 := func(msg []byte) {
+		log.Infof("received message 2: %s", string(msg))
+		log.Infof("len, %v", len(goldenMessage))
+		log.Infof("len, %v", len(msg))
+		if !bytes.Equal(msg, goldenMessage) {
+			log.Errorf("bad response")
+		}
+		assert.Equal(t, msg, goldenMessage)
+		done2 <- true
+	}
+
+	p1, _ := makeServerPeer(priv_key1, "Network1", 1, 0, func([]byte) {})
+	defer p1.Stop()
+	p2, _ := makeServerPeer(priv_key2, "Network1", 1, 0, handler1)
+	defer p2.Stop()
+	p3, _ := makeServerPeer(priv_key3, "Network1", 1, 0, handler2)
+	defer p3.Stop()
+
+	// Connect 2 of them with asdditional validator channel
+	go p1.ConnectValidatorChannel("peer 1")
+	go p2.ConnectValidatorChannel("peer 2")
+
+	// p1.BroadcastMessage("netwokr")
+
+	// Send Message from V for NV
+	// Broadcast non-validator message from one validator,  verify all peers has received it
+	// p1.BroadcastMessage(goldenMessage)
```

Comment:
We need only one broadcast and after that all peers should receive it

Reply:
I think, We will be needing two broadcasts:
1. for validator channel
2. for non validator channels